functional/FunctionalTest.java

```diff
@@ -2070,9 +2079,18 @@ public static void setBucketPolicy_test1() throws Exception {
       mintSuccessLog("setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)",
                      null, startTime);
     } catch (Exception e) {
-      mintFailedLog("setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)", null, startTime,
-                    null, e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
-      throw e;
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse == null || errorResponse.errorCode() != ErrorCode.NOT_IMPLEMENTED) {
+        mintFailedLog("setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)", null, startTime,
+                      null, e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
```

Comment:
Why is the NOT_IMPLEMENTED return code checked only for the case where you are setting the policy type to PolicyType.None, why not in the other scenarios when calling the SetBucketPolicy API?

Reply:
Done. Fixed.